Source/engine/point.hpp

```diff
@@ -131,6 +155,20 @@ struct Point {
 		return (approx + 512) / 1024;
 	}
 
+	/**
+	 * @brief Calculates the exact distance between two points (as accurate as the closest integer representation)
+	 *
+	 * In practice it is likely that ApproxDistance gives the same result, especially for nearby points.
+	 * @param other Point to which we want the distance
+	 * @return Exact magnitude of vector this -> other
+	*/
+	int ExactDistance(Point other) const
+	{
+		auto vector = *this - other; //No need to call abs() as we square the values anyway
+
+		return sqrt(vector.x * vector.x + vector.y * vector.y);

```

Comment:
`#include <algorithm>`

Reply:
<algorithm> is already included by stdcompat/algorithm.hpp, I thought stdcompat/abs.hpp was including <cmath> but that has <cstdlib> instead.